syntax = "proto3";

package paroduk;

message Produk {
  int32 id = 1;
  string nama = 2;
  string deskripsi = 3;
  float harga = 4;
  string gambar = 5;
  int32 stok = 6;
}

message ProdukListRequest {
  
}

message ProdukListResponse {
  repeated Produk produk = 1;
}

message ProdukDetailRequest {
  int32 id = 1;
}

message ProdukDetailResponse {
  Produk Produk = 1;
}

message ProdukCreateRequest {
  string nama = 1;
  string deskripsi = 2;
  float harga = 3;
  string gambar = 4;
  int32 stok = 5;
}

message ProdukCreateResponse {
  Produk produk = 1;
}

message ProdukUpdateRequest {
  int32 id = 1;
  string nama = 2;
  string deskripsi = 3;
  float harga = 4;
  string gambar = 5;
  int32 stok = 6;
}

message ProdukUpdateResponse {
  Produk Produk = 1;
}

message ProdukDeleteRequest {
  int32 id = 1;
}

message ProdukDeleteResponse {
  
}

message HitungHargaRequest {
  repeated int32 id = 1;
}

message HitungHargaResponse {
  float total = 1;
}

service ProdukService {
  rpc List(ProdukListRequest) returns (ProdukListResponse);
  rpc Detail(ProdukDetailRequest) returns (ProdukDetailResponse);
  rpc Create(ProdukCreateRequest) returns (ProdukCreateResponse);
  rpc Update(ProdukUpdateRequest) returns (ProdukUpdateResponse);
  rpc Delete(ProdukDeleteRequest) returns (ProdukDeleteResponse);
  rpc HitungHarga(HitungHargaRequest) returns (HitungHargaResponse);
}
